### PR TITLE
fixing error when istgt is started with no volume

### PR DESCRIPTION
**Pull request by mynktl**
Changes : 
- If `ISTGT` is started without any volume configuration then replication module will not accept connection from a replica.
- In `replication_test`, if it fails to make a connection with the target then it will exit with return value `1`.

### CODE REVIEW

File: src/istgt_lu.h [hunk 1]
```diff
@@ -326,6 +328,7 @@ typedef struct istgt_lu_t {
 	ISTGT_LU_MAP map[MAX_LU_MAP];
 	int conns;
 #ifdef REPLICATION
+	uint64_t replica_id[MAXREPLICA];
```
Comment (vishnuitta) [inline src/istgt_lu.h]: why is this required?

File: src/replication.c [hunk 2]
```diff
@@ -1453,6 +1453,12 @@ accept_mgmt_conns(int epfd, int sfd)
                 }
                 MTX_UNLOCK(&specq_mtx);
 
+		if (!spec) {
+			REPLICA_ERRLOG("Spec is not configured\n");
+			close(mgmt_fd);
```
Comment (vishnuitta) [inline src/replication.c]: you need to do 'shutdown' on fd also to inform other side of fd

File: src/replication_test.c [hunk 3]
```diff
@@ -550,6 +550,11 @@ main(int argc, char **argv)
 					}
 					break;
 				}
+				if (count == 0) {
```
Comment (vishnuitta) [inline src/replication_test.c]: are we hitting this when just 'close' is done on other side?

File: src/replication_test.c [hunk 4]
```diff
@@ -614,6 +622,9 @@ main(int argc, char **argv)
 						} else if(((uint64_t)count < total_len - recv_len) && errno == EAGAIN) {
 							recv_len += count;
 							break;
+						} else if (count == 0) {
```
Comment (vishnuitta) [inline src/replication_test.c]: this should be after initial 'if' check for count < 0..
same thing with next chunk of changes also

**Review by vishnuitta — COMMENTED** [review]
testcase need to be added, like,
- bring up with no LU
- connect with replicas
- add LU section and do refresh
- connect should be done automatically

File: src/istgt_lu.h (hunk 1 above)
Comment (mynktl) [reply src/istgt_lu.h]: ok. My base branch was different. That's why this change is here. I will remove this change.

File: src/replication.c (hunk 2 above)
Comment (mynktl) [reply src/replication.c]: ok

File: src/replication_test.c (hunk 3 above)
Comment (mynktl) [reply src/replication_test.c]: No.  Actually, in `test_read_data`, we are not handling return value `0` from `read`. Due to this, have added this check in the caller of `test_read_data`.

File: src/replication_test.c (hunk 4 above)
Comment (mynktl) [reply src/replication_test.c]: ok

**Review by vishnuitta — APPROVED** [review]
changes are good
